nixos/modules/programs/captive-browser.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.captive-browser;
+
+  package = pkgs.buildGoPackage rec {
+    name = "captive-browser";
+    version = "2018-11-08";
+    goPackagePath = name;
+
+    src = pkgs.fetchFromGitHub {
+      owner  = "FiloSottile";
+      repo   = "captive-browser";
+      rev    = "0f84599ddb36c090cf000681f86321aacb354c4a";
+      sha256 = "0fw4g4fsj9gwm711yj81z64bnmfj2g7wfkhnlsj1nfav0n5kfh40";
+    };
+  };
+in
+{
+  ###### interface
+
+  options = {
+    programs.captive-browser = {
+      enable = mkEnableOption "captive browser";
+
+      wlan-interface = mkOption {
+        type = types.str;
+        description = "your public network interface (wlp3s0, wlan0, eth0, ...)";
+      };
+
+      # the options below are the same as in "captive-browser.toml"
+      browser = mkOption {
+        type = types.str;
+        default = concatStringsSep " " [ ''${pkgs.chromium}/bin/chromium''
```

Comment:
Maybe good enough for now, but would be nice if it wasn't chromium-specific by default. :) Worth making it conditional if google-chrome is installed instead? (Though there's also google-chrome-{dev,beta})

Reply:
Yes, changing the browser (even within chromium versions, but there is also Opera, Firefox, ..) might require adjusting the command line switches too.
So I am not sure if it worth to separate options for the browser executable and the command line switches.

A good candidate default would be the lightweight browser possible (lynx?)

Reply:
Hm, I'd say about 1/3 of the captive portals I run into require javascript. For example, gogoinflight/aainflight/etc.

Could build a tiny shim browser that uses the OS's native webview, maybe upstream it to captive-browser. I'll look into that. :)

Reply:
Same here regarding `defaultText`.